pkg/virt-api/api.go

```diff
@@ -373,6 +379,10 @@ func (app *virtAPIApp) composeSubresources() {
 						Name:       "virtualmachineinstances/console",
 						Namespaced: true,
 					},
+					{
+						Name:       "virtualmachineinstances/tunnel",
```

Comment:
Do you have an understanding for why your method for tunneling through the api server works, but attempting to do a `kubectl port-forward` fails? I'm not sure I grasp why the port-forward doesn't work

If we add the ability to establish connections like this to a VMI across all ports, I think we'll need to document how this differs from port-forwarding and allows for use cases that wouldn't be possible otherwise.

Reply:
My unconfirmed theory still is that it's the way port-forwarded traffic looks to the virt-manager/VM as it comes from the node/kubelet somehow.
The tunnel traffic is looking like normal external traffic from inside the cluster instead.

I thought about documentation as well. It feels like we should at least recommend not using it for arbitrary traffic as it would put strain on the apiserver instead the actual networking parts that are intended for production traffic.

Reply:
The issue is that `port-forward` forwards `localhost` ports from the pods. Our VMs however only look at the main interface. I discussed with @phoracek a few ways on how to address this.

There exist a few possibilities: 
 * allow adding a second interface to the VMI which is effectively connected to localhost in the pod network namespace and mirrors the traffic
  * some explicit proxying with virt-launcher
  * iptables/tc magic to try to combine both interfaces (like done for istio support in the past which also works with localhost)

Reply:
This should be fixed now. @alonSadan and @rhrazdil  have resolved it. I believe it was via https://github.com/kubevirt/kubevirt/pull/5431.

Reply:
> This should be fixed now. @alonSadan and @rhrazdil have resolved it. I believe it was via #5431.

Awesome. This means that we potentially don't need an extra endpoint at all and can just use the pods portforward.

Reply:
Does that work for all pod network bindings or just masquerade?

Reply:
port-forward in Kubernetes also only supports TCP while what we have here should also support UDP connections.